pmm-app/src/pmm-qan/panel/components/Details/Metrics/Metrics.utils.ts

```diff
@@ -1,33 +1,30 @@
-import {
-  applyFieldOverrides, DataFrame, FieldType, GrafanaTheme2, toDataFrame,
-} from '@grafana/data';
-import { HistogramResponse } from './Metrics.types';
-
-export const histogramToDataFrame = ({
-  histogram_items = [],
-}: HistogramResponse,
-theme: GrafanaTheme2): DataFrame[] => {
+import { ChartData } from 'chart.js';
+import { GrafanaTheme2 } from '@grafana/data';
+import { HistogramAPI } from './Metrics.types';
+
+export const getChartDataFromHistogramItems = (histogram_items: HistogramAPI[], theme: GrafanaTheme2): ChartData<'bar'>| undefined => {
+  const total = histogram_items ? histogram_items.reduce(
+    (previousValue, currentValue) => (
+      currentValue.frequency ? previousValue + currentValue.frequency : previousValue
+    ), 0,
+  ) : null;
+
   const ranges = histogram_items.map(({ range }) => formatRange(range));
   const frequencies = histogram_items.map(({ frequency }) => frequency || 0);
-  const fields = {
-    fields: [
-      { name: 'x', type: FieldType.string, values: ranges },
-      { name: '', type: FieldType.number, values: frequencies },
-    ],
+  const dataInPersent = total !== null
+    ? histogram_items.map(({ frequency }) => Math.round((frequency || 0 / total) * 10000) / 100)
+    : histogram_items.map(() => undefined);
```

Comment:
Isn't enough to return undefined or empty array? Do we really need to put each value as undefined?

Reply:
No, it's not necessary. I think I can handle this moment elsewhere, so I'll fix it.